lib/tbd/psi.rb

```diff
@@ -1598,97 +1784,193 @@ def process(model = nil, argh = {})
+        is[:balconysill ] = set.keys.to_s.include?("balconysill")
+        is[:rimjoist    ] = set.keys.to_s.include?("rimjoist")
+
+        # Label edge as ...
+        #         :head,         :sill,         :jamb (vertical fenestration)
+        #     :doorhead,     :doorsill,     :doorjamb (opaque door)
+        # :skylighthead, :skylightsill, :skylightjamb (all other cases)
+        #
+        # ... if linked to:
+        #   1x subsurface (vertical or non-vertical)
         edge[:surfaces].keys.each do |i|
-          break    if is[:head] || is[:sill] || is[:jamb]
+          break    if is[:head        ]
+          break    if is[:sill        ]
+          break    if is[:jamb        ]
+          break    if is[:doorhead    ]
+          break    if is[:doorsill    ]
+          break    if is[:doorjamb    ]
+          break    if is[:skylighthead]
+          break    if is[:skylightsill]
+          break    if is[:skylightjamb]
           next     if deratables.include?(i)
           next unless holes.key?(i)
 
-          gardian = ""
-          gardian = id if deratables.size == 1 # just dad
-
-          if gardian.empty? # seek uncle
-            pops   = {} # kids?
-            uncles = {} # nieces?
-            boys   = [] # kids
-            nieces = [] # nieces
-
-            uncle  = deratables.first unless deratables.first == id # uncle #1?
-            uncle  = deratables.last  unless deratables.last  == id # uncle #2?
-
-            pops[:w  ] = tbd[:surfaces][id   ].key?(:windows)
-            pops[:d  ] = tbd[:surfaces][id   ].key?(:doors)
-            pops[:s  ] = tbd[:surfaces][id   ].key?(:skylights)
-            uncles[:w] = tbd[:surfaces][uncle].key?(:windows)
-            uncles[:d] = tbd[:surfaces][uncle].key?(:doors)
-            uncles[:s] = tbd[:surfaces][uncle].key?(:skylights)
-
-            boys   += tbd[:surfaces][id   ][:windows  ].keys if   pops[:w]
-            boys   += tbd[:surfaces][id   ][:doors    ].keys if   pops[:d]
-            boys   += tbd[:surfaces][id   ][:skylights].keys if   pops[:s]
-            nieces += tbd[:surfaces][uncle][:windows  ].keys if uncles[:w]
-            nieces += tbd[:surfaces][uncle][:doors    ].keys if uncles[:d]
-            nieces += tbd[:surfaces][uncle][:skylights].keys if uncles[:s]
-
-            gardian = uncle if   boys.include?(i)
-            gardian = id    if nieces.include?(i)
+          # In most cases, subsurface edges simply delineate the rough opening
+          # of its base surface (here, a "gardian"). Door sills, corner windows,
+          # as well as a subsurface header aligned with a plenum "floor"
+          # (ceiling tiles), are common instances where a subsurface edge links
+          # 2x (opaque) surfaces. Deratable surface "id" may not be the gardian
+          # of subsurface "i" - the latter may be a neighbour. The single
+          # surface to derate is not the gardian in such cases.
+          gardian = deratables.size == 1 ? id : ""
```

Comment:
Logic slightly re-written (and now somewhat better documented). Largely due to 90.1 distinction of "vertical fenestration" (vs other subsurface classification).